packages/editor/src/components/autosave-monitor/index.js

```diff
@@ -6,63 +6,53 @@ import { compose } from '@wordpress/compose';
 import { withSelect, withDispatch } from '@wordpress/data';
 
 export class AutosaveMonitor extends Component {
-	componentDidUpdate( prevProps ) {
-		const {
-			isDirty,
-			editsReference,
-			isAutosaveable,
-			isAutosaving,
-		} = this.props;
+	constructor( props ) {
+		super( props );
+		this.needsAutosave = !! ( props.isDirty && props.isAutosaveable );
+	}
 
-		// The edits reference is held for comparison to avoid scheduling an
-		// autosave if an edit has not been made since the last autosave
-		// completion. This is assigned when the autosave completes, and reset
-		// when an edit occurs.
-		//
-		// See: https://github.com/WordPress/gutenberg/issues/12318
+	componentDidMount() {
+		this.setAutosaveTimer();
+	}
 
-		if ( editsReference !== prevProps.editsReference ) {
-			this.didAutosaveForEditsReference = false;
+	componentDidUpdate( prevProps ) {
+		if ( ! this.props.isDirty && prevProps.isDirty ) {
+			this.needsAutosave = false;
+			return;
 		}
 
-		if ( ! isAutosaving && prevProps.isAutosaving ) {
-			this.didAutosaveForEditsReference = true;
+		if ( this.props.isAutosaving && ! prevProps.isAutosaving ) {
+			this.needsAutosave = false;
+			return;
 		}
 
-		if (
-			prevProps.isDirty !== isDirty ||
-			prevProps.isAutosaveable !== isAutosaveable ||
-			prevProps.editsReference !== editsReference
-		) {
-			this.toggleTimer(
-				isDirty && isAutosaveable && ! this.didAutosaveForEditsReference
-			);
+		if ( this.props.editsReference !== prevProps.editsReference ) {
+			this.needsAutosave = true;
 		}
 	}
 
 	componentWillUnmount() {
-		this.toggleTimer( false );
+		clearTimeout( this.timerId );
 	}
 
-	toggleTimer( isPendingSave ) {
-		const { interval, shouldThrottle = false } = this.props;
+	setAutosaveTimer( timeout = this.props.interval * 1000 ) {
+		this.timerId = setTimeout( () => {
+			this.autosaveTimerHandler();
+		}, timeout );
+	}
 
-		// By default, AutosaveMonitor will wait for a pause in editing before
-		// autosaving. In other words, its action is "debounced".
-		//
-		// The `shouldThrottle` props allows overriding this behaviour, thus
-		// making the autosave action "throttled".
-		if ( ! shouldThrottle && this.pendingSave ) {
-			clearTimeout( this.pendingSave );
-			delete this.pendingSave;
+	autosaveTimerHandler() {
+		if ( ! this.props.isAutosaveable ) {
+			this.setAutosaveTimer( 1000 );
```

Comment:
Is this fixed `1000` number correct? Shouldn't this take "interval" into consideration?

Reply:
It could be either `1000` or `interval` depending on how you look at it. Was there a bug report related to this?

A case for `1000`:
* We are not autosaveable and, from the perspective of this component, we don't know why – could be save in progress, could be internet connectivity issues
* We want to know when we are saveable again as soon as possible because the `interval` could be 30 seconds and we could be intermittently non-autosaveable again on the next run

A case for `interval`:
* The handler explicitly asked the timer to run at most once per `interval` seconds
* The handler is responsible for handling `isAutosaveable` property so it should already know if we missed an `autosave()` call

Reply:
No, no bug, just looked weird. I was working on something unrelated requiring disabling autosave when I found it. Thanks for the explanation, it makes sense.